builtin/rebase.c

```diff
@@ -376,6 +377,19 @@ static int run_rebase_interactive(struct rebase_options *opts,
 	flags |= opts->rebase_cousins > 0 ? TODO_LIST_REBASE_COUSINS : 0;
 	flags |= command == ACTION_SHORTEN_OIDS ? TODO_LIST_SHORTEN_IDS : 0;
 
+	if (opts->ignore_whitespace) {
+		struct strbuf buf = STRBUF_INIT;
+
+		if (!opts->strategy)
+			opts->strategy = "recursive";
+		if (opts->strategy_opts)
+			strbuf_addstr(&buf, opts->strategy_opts);
+
+		strbuf_addstr(&buf, " ignore-space-change");
+		opts->strategy_opts = xstrdup(buf.buf);
+		strbuf_release(&buf);
+	}
```

Comment:
> This block looks an awful lot like the "if (strategy_options.nr)" block in cmd_rebase(), except that it puts a double dash before each option. Why does it use a double dash and you don't?

I wanted to make the code more coupled with `rebase -i`. What I mean is, let common rebase do the preprocessing then make the necessary changes when and if the time comes. Also, adding `--` does not change anything since we are removing them anyway (see [sequencer.c:2480 void parse_strategy_opts](https://github.com/r1walz/git/blob/aa25c82427ae70aebf3b8f970f2afd54e9a2a8c6/sequencer.c#L2480)). But to make code consistent, I can add `--`.

> Would it make sense to just append to strategy_options before that other block instead of needing to have a similar block of code here?

Also, I didn't want to change `strategy_options.nr` manually.

Reply:
We manually add the double dashes and then remove them?  That seems rather Rube Goldberg-esque.  But that's getting outside the scope of your patch.  I'd say use the double dash for consistency.

Why would you change strategy_options.nr manually?  That would seem kind of odd.  I would think it'd be more of a

if (options.type != REBASE_AM)
    string_list_append(&strategy_options, " --ignore-space-change");

after the setting of the default rebase type (i.e. outside the block a few lines after '/* the default rebase backend is `--am` */'), and coupled with moving the 'if (strategy_options.nr)' block down to be after these two new lines.  That would allow you to drop this whole block here...*if* it's actually okay to move the 'if (strategy_options.nr)' block down.

Reply:
The problem with this is: what if the user use `git rebase--interactive --ignore-whitespace --onto master --switch-to topic`? Since we are not handling --ignore-whitespace on the rebase--interactive side, this will result in a bug. So, let `do_interactive_rebase` or `run_rebase_interactive` do this task?

Reply:
Ah, good catch.  I was indeed overlooking the rebase--interactive command.